Linux/First_task/copy-dir.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+SOURCE_DIR=$(realpath "$1")
+DESTINATION_DIR=$(realpath "$2")
+DATE_FORMAT=$(date +%Y%m%d_%H%S)
+
+the_same_names () {
+  if [ "$1" == "$2" ]
+  then
+    echo "The same names"
+    exit 0
+  fi
+}
+
+#Free disk
+get_free_disk_space () {
+  df "$1" | tail -1 | awk '{print$4}'
+}
+
+directoryproblem () {
+  PARENT_SOURCE=$(dirname "$1")
+  PARENT_DESTINATION=$(dirname "$2")
+
+  if [ "$1" == "$PARENT_DESTINATION" ]
+  then
+    echo ""$2" parent directory "$1""
+    exit 0
+  elif [ "$2" == "$PARENT_SOURCE" ]
+  then 
+    echo ""$1" parent directory "$2""
+  fi
+}
+
+#Volume directory
+get_dir_space () {
+  du -s "$1" | awk '{print$1}'
+}
+
+#Copy direct
+copyall () {
+  cp -p -r $1 $2
+}
+
+copy_archive () {
```

Comment:
This function does not copy the archive. Why is it named "copy_archive"? Perhaps a better name would be "archive_and_cleanup" since the function is also responsible for deleting old archives.

Reply:
The algorithm responsible for removing outdated archives should ideally be implemented in a separate function